autoload/vader/assert.vim

```diff
@@ -53,8 +53,13 @@ endfunction
 function! vader#assert#equal(...)
   let [exp, got] = a:000[0:1]
   let s:assertions[1] += 1
+  if exp is 0 || got is 0
+      let l:comparison = exp isnot# got
+  else
+      let l:comparison =  exp !=# got
+  endif
 
-  if exp !=# got
+  if l:comparison
```

Comment:
IIRC I have done this locally, and then used another error message in that case.